### PR TITLE
NSIDCbin:  Sea Ice Concentrations new raster driver 

### PR DESCRIPTION
**Pull request by mdsumner**
New raster driver. 

Supported by GDAL for read access. This format is a raw binary format for the Nimbus-7 SMMR and DMSP SSM/I-SSMIS Passive Microwave Data sea ice concentrations. There are daily and monthly maps in the north and south hemispheres supported by this driver.

A simple test and example file is included.

Originally drafted at #6183, where this benefited enormously from review and feedback -  but has been recrafted here. 

 - [x] Add test case(s)
 - [x] Add documentation
 - [x] Updated Python API documentation (swig/include/python/docs/)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Comment by rouault** [comment]
@mdsumner I strongly sugges you setup pre-commit to solve all the linting issues: https://gdal.org/development/dev_practices.html#commit-hooks

File: autotest/gdrivers/nsidcbin.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+###############################################################################
+# Perform simple read test.
+
+## test file
+## mkdir autotest/gdrivers/data/nsidcbin
+## cd autotest/gdrivers/data/nsidcbin
+## wget ftp://sidads.colorado.edu/pub/DATASETS/nsidc0081_nrt_nasateam_seaice/south/nt_20220409_f18_nrt_s.bin
+## ## /vsicurl/ftp://sidads.colorado.edu/pub/DATASETS/nsidc0081_nrt_nasateam_seaice/south/nt_20220409_f18_nrt_s.bin
+def test_nsidcbin_1(filename='data/nsidcbin/nt_20220409_f18_nrt_s.bin'):
+   ds = gdal.Open(filename)
```
Comment (rouault) [inline autotest/gdrivers/nsidcbin.py:47]:
```suggestion
    ds = gdal.Open(filename)
```

**Comment by mdsumner** [comment]
I understand, I'm having problems with setting it up 🙏 - i need to spend some time getting my env sorted

**Comment by rouault** [comment]
> I'm having problems with setting it up pray

Same error as in https://github.com/OSGeo/gdal/pull/7253#discussion_r1109164972 ?

If there are lessons to learn from the problems you get, we could document them. The hope was the pre-commit was supposed to be a no-brainer for contributors...

**Comment by mdsumner** [comment]
yes, that exactly - I found that stackoverflow answer  -  installing isort from github source doesn't fix (but I am also very noob at python installs and problem solving there)

fwiw,  and as much for my reference as I figure it out: 

python 3.8.10   ## I'm  using python3 -m pip install ...
isort 5.12.0
poetry 1.3.2
pre-commit 3.0.4


The linting *was* very easy on my old setup. If I get an understanding I'll definitely share it. I can use docker or conda etc but I'm stuck atm. 

Got a number of little PRs in the waiting too :)

**Comment by mdsumner** [comment]
ok good news, modifying .pre-commit-config.yaml did the job

I'm learnin' ... :)

File: autotest/gdrivers/nsidcbin.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+
+from osgeo import gdal
```
Comment (rouault) [inline autotest/gdrivers/nsidcbin.py:32]:
```suggestion
import pytest

from osgeo import gdal

pytestmark = pytest.mark.require_driver("NSIDCbin")
```

**Pull request merged**